### PR TITLE
Adding query arguments for the BuildImageConfiguration

### PR DESCRIPTION
**Pull request by weiserr**
The changes within this PullRequest allow adding arbitrary "simple" queryArguments to the Docker daemon build REST interface (https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/build-image-from-a-dockerfile).

Currently no changes have been performed to docs/tests other than fixing compile errors. If the changes within this PullRequest are acceptable I will extend and squash it.

### CODE REVIEW

**Comment by rhuss** [comment]
Thanks for the PR ! Could you briefly describe your use case, e.g. what kind of query parameters to you wanto to send ?

**Comment by weiserr** [comment]
Hi Roland

we are required to set the shared memory and memory settings when building the container.

File: src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java [hunk 1]
```diff
@@ -235,6 +240,10 @@ public BuildTarArchiveCompression getCompression() {
         return compression;
     }
 
+    public Map<String, String> getQueryArgs() {
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: Maybe we can call it some what less technical (e.g. how it is used) but e.g. 'options' (or 'buildOptions') ?

Comment (weiserr) [inline src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: `buildOptions` sounds good.

**Comment by weiserr** [comment]
Hi Roland - I have updated the PullRequest

**Comment by rhuss** [comment]
Thanks ! And very sorry for the long delay, busy times ;-( Next week I have some spare cycles and will integrate the PR then. You can also expect a release of the end next week, the christmas edition ;-)

**Comment by weiserr** [comment]
Hi Roland

thanks for the reply - happy to hear that. Do you need me to update the PullRequest to resolve the conflicts?

Cheers

**Comment by rhuss** [comment]
Ah, sorry. I messed it up as I changed our process to integration directly into master. Could you please recreate a new PR against master ? This would be the easiests way to get back on track ....

thanks and sorry for any inconvenience ....

**Comment by weiserr** [comment]
No worries - I have created another PullRequest: #666

**Comment by rhuss** [comment]
Awesome,thx unfortunately I just catched an ugly flu, and staying in bed
currently. I will continue here asap. Sorry ...

Robert Weiser <notifications@github.com> schrieb am Di., 20. Dez. 2016,
13:40:

**Comment by rhuss** [comment]
Closed in favor of #666

**Pull request closed without merging**